packages/ember-metal/lib/computed.js

```diff
@@ -327,13 +328,17 @@ ComputedPropertyPrototype.didChange = function(obj, keyName) {
 
 ComputedPropertyPrototype.get = function(obj, keyName) {
   if (this._volatile) {
-    return this._getter.call(obj, keyName);
+    return this._getter ? this._getter.call(obj, keyName) : undefined;
```

Comment:
I don't think it is best solution, since `this._getter` won't be provided in super narrow case, and in common case it is provided, the best solution is to provide empty function as you did earlier

Reply:
`noop` function for `_getter` in constructor is best solution I think

Reply:
Well, it depends. If we use the `noop` function in the constructor we will allocate a memory even if we never call the `get`.

Right, now the solution is more complex but it fixes the source of the problem and saves resources. I noticed that when I look at `set` function definition. Now, they are implemented in a similar way.

Do you think this is a bad approach?

Reply:
> Well, it depends. If we use the noop function in the constructor we will allocate a memory even if we never call the get.

thats why I asked to keep `noop` function outside constructor ([example](https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/lib/renderer.ts#L184)), also not sure if it should contain warning inside `warn('getter missing .....')` this should be confirmed by core team I guess

> Right, now the solution is more complex but it fixes the source of the problem and saves resources. I noticed that when I looked at set function definition. Now, they are implemented in a similar way.

Not only it is complex, it is also adds extra overhead/processing to be done each time even `getter` is provided (which is 99% of time)

Thats why your first approach was best

Reply:
it is hotpath code therefore slightly more work will result in big overheads

Reply:
You suggested: `noop function for _getter in constructor is best solution I think` not, outside :p

Ok, will try that one. But isn't Ember.K already deprecated? Shouldn't we just use `() {}`?

Reply:
> You suggested: noop function for _getter in constructor is best solution I think not, outside :p
https://github.com/emberjs/ember.js/pull/15962#discussion_r155918993

> Ok, will try that one. But isn't Ember.K already deprecated? Shouldn't we just use () {}? 

global `Ember.K` yes deprecated, but using `noop` function  naming `K` is not :P, but calling it `noop` is more descriptive I think